plugins/modules/openssl_pkcs12.py

```diff
@@ -438,17 +518,152 @@ def parse(self):
+
+    def _dump_other_certificates(self, pkcs12):
+        return [
+            crypto.dump_certificate(crypto.FILETYPE_PEM, other_cert)
+            for other_cert in pkcs12.get_ca_certificates()
+        ]
+
+    def _get_friendly_name(self, pkcs12):
+        return pkcs12.get_friendlyname()
+
+
+class PkcsCryptography(Pkcs):
+    def __init__(self, module):
+        super(PkcsCryptography, self).__init__(module, 'cryptography')
+
+        serialize_key_and_certificates,
+        load_key_and_certificates,
+
+    def generate_bytes(self, module):
+        """Generate PKCS#12 file archive."""
+        pkey = None
+        if self.privatekey_path:
+            try:
+                pkey = load_privatekey(self.privatekey_path, self.privatekey_passphrase, backend=self.backend)
+            except OpenSSLBadPassphraseError as exc:
+                raise PkcsError(exc)
+
+        cert = None
+        if self.certificate_path:
+            cert = load_certificate(self.certificate_path, backend=self.backend)
+
+        friendly_name = to_bytes(self.friendly_name) if self.friendly_name is not None else None
+
+        # Store fake object which can be used to retrieve the components back
+        self.pkcs12 = (pkey, cert, self.other_certificates, friendly_name)
+
+        return serialize_key_and_certificates(
+            friendly_name,
+            pkey,
+            cert,
+            self.other_certificates,
+            serialization.BestAvailableEncryption(to_bytes(self.passphrase))
+            if self.passphrase is not None else serialization.NoEncryption(),
+        )
+
+    def parse_bytes(self, pkcs12_content):
+        try:
+            private_key, certificate, additional_certificates = load_key_and_certificates(
+                pkcs12_content, self.passphrase)
+
+            pkey = None
+            if private_key is not None:
+                pkey = private_key.private_bytes(
+                    encoding=serialization.Encoding.PEM,
+                    format=serialization.PrivateFormat.TraditionalOpenSSL,
+                    encryption_algorithm=serialization.NoEncryption(),
+                )
+
+            crt = None
+            if certificate is not None:
+                crt = certificate.public_bytes(serialization.Encoding.PEM)
+
+            other_certs = []
+            if additional_certificates is not None:
+                other_certs = [
+                    other_cert.public_bytes(serialization.Encoding.PEM)
+                    for other_cert in additional_certificates
+                ]
+
+            friendly_name = None
+            if certificate:
+                # See https://github.com/pyca/cryptography/issues/5760#issuecomment-842687238
+                friendly_name = certificate._backend._ffi.string(
+                    certificate._backend._lib.X509_alias_get0(
+                        certificate._x509, certificate._backend._ffi.new("int *")
+                    )
+                )
```

Comment:
```suggestion
                maybe_name = certificate._backend._lib.X509_alias_get0(
                    certificate._x509, certificate._backend._ffi.new("int *")
                )
                if maybe_name != certificate._backend._ffi.NULL:
                     friendly_name = certificate._backend._ffi.string(maybe_name)
```
Since this is a C function we need a null check (or try block) here otherwise if alias is `NULL` (cannot be retrieved from the x509 struct) we get a runtime error like `RuntimeError: cannot use string() on <cdata 'unsigned char *' NULL>`.

Reply:
I was planning to look into that, but ran out of time yesterday and really had to go to bed ;-) Thanks for the suggestion, I've committed it.